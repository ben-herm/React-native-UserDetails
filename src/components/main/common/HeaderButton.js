import React from 'react'
import {StyleSheet, TouchableOpacity, Platform} from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'

class HeaderButton extends React.Component {
  static navigationOptions = ({navigation}) => {
    return {
      title: navigation.getParam('otherParam', 'A Nested Details Screen'),
    }
  }

  render () {
    return (
      <TouchableOpacity
        style={{
          paddingHorizontal: 25,
          marginTop: Platform.OS == 'ios' ? 0 : 20,
        }}
        onPress={() => this.props.navigation.pop()}
        title='Info'
        color='white'>
        <Icon name='arrow-circle-left' size={30} color='#900' />
      </TouchableOpacity>
    )
  }

  /* render function, etc */
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'black',
    flex: 1,
    flexDirection: 'column',
  },
})

export default HeaderButton
