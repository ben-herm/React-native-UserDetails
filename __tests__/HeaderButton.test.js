import React from 'react';
import HeaderButton from '../src/components/main/common/HeaderButton';
import renderer from 'react-test-renderer';

const createTestProps = (props) => ({
  navigation: {
    getParam: jest.fn(),
  },
  ...props,
});

describe('HeaderButton rendering', () => {
  const navigationMock = {navigation: {getParams: {}}};
  it('should render with navigation', () => {
    const rendered = renderer
      .create(<HeaderButton navigation={navigationMock} />)
      .toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
});

describe('renders correctly', () => {
  it('renders correctly', () => {
    renderer.create(<HeaderButton />);
  });
});
