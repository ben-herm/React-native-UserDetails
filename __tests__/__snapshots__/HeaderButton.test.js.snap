// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton rendering should render a <View /> 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginTop": 0,
      "opacity": 1,
      "paddingHorizontal": 25,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#900",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`HeaderButton rendering should render with navigation 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginTop": 0,
      "opacity": 1,
      "paddingHorizontal": 25,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#900",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
